wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/tool/kubernetes/KubernetesToolToWorkspaceApplier.java

```diff
@@ -128,8 +139,10 @@ private String retrieveContent(Tool recipeTool, @Nullable FileContentProvider fi
     return recipeFileContent;
   }
 
-  private List<HasMetadata> filter(KubernetesList list, Map<String, String> selector) {
-    return list.getItems().stream().filter(item -> matchLabels(item, selector)).collect(toList());
+  private List<HasMetadata> filter(List<HasMetadata> list, Map<String, String> selector) {
```

Comment:
Could you rename the method to make its purpose more intuitive?

Reply:
Which name looks better here for you? `filterByLabels`
Maybe we can inline this method and then it would look like the following:
```java
    ...
    Map<String, String> selector = k8sTool.getSelector();
    if (!selector.isEmpty()) {
      list = list.stream()
          .filter(item -> matchLabels(item, selector))
          .collect(toCollection(ArrayList::new));
    }
   ...
```

Reply:
I would not inline it. But it is not clear at all why we need that method. And filter by labels tells what but doesn't tell why. Maybe `removeNotMatchingSelector`?

Reply:
`removeNotMatchingSelector` looks confusing for me. Because `Collection#remove`/`Collection#removeAll`/`Collection#removeIf` removes matched objects from collection and returns removed objects.
From `removeNotMatchingSelector` I would expect that it modified the specified list and returns elements which are not matched with a selector.
Added a java doc to filter methods, maybe it will be good enough.

Reply:
:+1: